### PR TITLE
Set up the database with basic functionality

### PR DESCRIPTION
**Pull request by BeardOfDan**
The app connects to our remote database (don't forget that since the .env file is included in .gitignore, you have to manually add it to the project each time you pull from master). There are 3 main functions for the database: save, loadAll, and loadParticular. 

For each of them, the first argument is the type of model ('student', 'superlative', 'event', or 'shoutout') the function is being used on. 

The second argument for save is the data for the model. 

loadAll only takes the one argument, since it loads all of the instances of that model type. 

loadParticular accepts an object for the second argument. That argument is a key value pair for identifying the particular model instance that you want to load. 

There are examples and further documentation in the comments in the code

Also, the server.js file has example paths that allow you to use the database functions.

### CODE REVIEW

**Comment by BeardOfDan** [comment]
Sorry about how long the pull request is. I waited until I had all of the listed content working before I did a pull request, I definitely should have broken that up across several pull requests.

File: server.js [hunk 1]
```diff
@@ -26,6 +28,153 @@ app.use(webpackDevMiddleware(compiler, {
...
+});
+
+// =========================
+// ===== Student Paths =====
+// =========================
+
+app.post('/addStudent', (req, res, next) => {
+  db.save('student', req.body.student)
+    .then((data) => {
+      res.end('You saved a new student with the data: ' + JSON.stringify(data, undefined, 2));
```
Comment (philcho) [inline server.js:48]: 2 spaces! lol

File: db/index.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+const SuperlativePage = mongoose.model('SuperlativePage', SuperlativePageSchema);
+const EventPage = mongoose.model('EventPage', EventPageSchema);
+const ShoutoutsPage = mongoose.model('ShoutoutsPage', ShoutoutsPageSchema);
+
+// ============================
+// ===== Helper Functions =====
+// ============================
+
+// a helper function to use the correct model
+// returns a model instance with the data
```
Comment (philcho) [inline db/index.js:75]: Great job with the comments.

File: db/index.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    return new EventPage(data);
+  } else if (modelType === 'shoutout') {
+    return new ShoutoutsPage(data);
+  } else {
+    console.log('ERROR! This is an unhandled case in the db -> index.js -> getModel');
+  }
+};
+
+// returns a the model for the functionality
+const getModelType = (modelType) => {
```
Comment (philcho) [inline db/index.js:91]: I like these helper functions.

File: db/index.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    .catch((e) => {
+      console.log('ERROR!\n  In db.save() when saving a model of type', modelType, '\n', e);
+    });
+};
+
+// ======================
+// === Load Functions ===
+// ======================
+
+// identifyier is an object that has the unique key/value pair
```
Comment (philcho) [inline db/index.js:126]: Minor typo ("identifyier"). You can change it next time. Or not. But I've learned that you care about these things ;)

File: server.js [hunk 5]
```diff
@@ -26,6 +28,153 @@ app.use(webpackDevMiddleware(compiler, {
...
+
+app.get('/getAllEvents', (req, res, next) => {
+  db.loadAll('event')
+    .then((data) => {
+      res.end('Here is all of our events: \n' + JSON.stringify(data, undefined, 2));
+    })
+});
+
+app.post('/getParticularEvent', (req, res, next) => {
+  // example postman request data
```
Comment (philcho) [inline server.js:122]: These examples are helpful, thanks.

**Review by philcho — APPROVED** [review]

**Pull request merged**